src/common/components/video-gallery/index.tsx

```diff
@@ -0,0 +1,431 @@
+import React, { useEffect, useState } from "react";
+import LinearProgress from "../linear-progress";
+import { informationVariantSvg } from "../../img/svg";
+import { Modal } from "react-bootstrap";
+import { _t } from "../../i18n";
+import "./index.scss";
+import DropDown from "../dropdown";
+import Tooltip from "../tooltip";
+import { ConfirmNsfwContent } from "../video-nsfw";
+
+interface videoProps {
+  status: string;
+  owner: string;
+  thumbUrl: string;
+  permlink: string;
+}
+
+interface Props {
+  showGallery: boolean;
+  setShowGallery: React.Dispatch<React.SetStateAction<boolean>>;
+  checkStat: () => Promise<any>;
+  insertText: (before: string, after?: string) => void;
+  setVideoEncoderBeneficiary?: (video: any) => void;
+  toggleNsfwC?: () => void;
+}
+
+const VideoGallery = (props: Props) => {
+  const {
+    showGallery,
+    setShowGallery,
+    checkStat,
+    insertText,
+    setVideoEncoderBeneficiary,
+    toggleNsfwC
+  } = props;
+
+  const [loading, setLoading] = useState(false);
+  const [items, setItems] = useState([]);
+  const [showMoreInfo, setShowMoreInfo] = useState(false);
+  const [hoveredItem, setHoveredItem] = useState<any>(null);
+  const [isEmbedded, setIsembedded] = useState(false);
+  const [label, setLabel] = useState("All");
+  const [filtered, setFiltered] = useState<any>([]);
+  const [isFiltered, setIsFiltered] = useState(false);
+
+  useEffect(() => {
+    getAllStatus();
+  }, [showGallery]);
+
+  useEffect(() => {
+    setIsFiltered(true);
+  }, [label]);
+
+  const getAllStatus = async () => {
+    setLoading(true);
+    const data = await checkStat();
+    if (data) {
+      setItems(data);
+      setLoading(false);
+      setLabel("All");
+    }
+  };
+
+  const setBeneficiary = (video: any) => {
+    setVideoEncoderBeneficiary && setVideoEncoderBeneficiary(video);
+  };
+
+  const formatTime = (dateStr: string | number | Date) => {
+    const date: any = new Date(dateStr);
+    const now: any = new Date();
+
+    const difference = Math.abs(now - date);
+    const seconds = Math.floor(difference / 1000);
+    const minutes = Math.floor(seconds / 60);
+    const hours = Math.floor(minutes / 60);
+    const days = Math.floor(hours / 24);
+    const months = Math.floor(days / 30);
+    const years = Math.floor(months / 12);
+
+    if (years > 0) {
+      if (years === 1) {
+        return `${years} year ago`;
```

Comment:
There is helper or function with this, why this is being handled manually? Check `dateToRelative` or `parse-date.ts` file.